marc.go

```diff
@@ -356,3 +359,39 @@ func TranslateLanguageCodes(recordCodes []string, languageCodes map[string]strin
 	}
 	return languages
 }
+
+// getLinks take a MARC record and eturns an array of Link objects from the 856 field data.
+func getLinks(marcrecord *marc21.Record) []Link {
+	var links []Link
+	marc856 := marcrecord.GetFields("856")
+	if len(marc856) == 0 {
+		return nil
+	}
+	for _, f := range marc856 {
+		switch f := f.(type) {
+		case *marc21.DataField:
+			ind1 := string(f.Ind1)
+			ind2 := string(f.Ind2)
+
+			if ind1 == "4" && (ind2 == "0" || ind2 == "1") {
+				link := Link{
+					Kind:         getSubfield(f, "3"),
+					URL:          getSubfield(f, "u"),
+					Text:         getSubfield(f, "y"),
+					Restrictions: getSubfield(f, "z")}
+				links = append(links, link)
+			}
+		}
+	}
+	return links
+}
+
+func getSubfield(field *marc21.DataField, code string) string {
```

Comment:
I think this method is a bit redundant to `stringifySelectSubfields` which I'm not surprised you didn't use based on it's less-than-:rainbow: name.

But I think if you call it like `Kind:         stringifySelectSubfields(f, []byte("3"))` you'll get the same results as this method.